### PR TITLE
Add validation for a few string fields

### PR DESCRIPTION
**Pull request by yiminc**
## What changed?
Add string validation for a few string fields

## Why?
Since we disable utf8 string validation from proto level, we want to enforce minimal validation for some key fields.

## How did you test it?
Unit tests

## Potential risks
No

### CODE REVIEW

File: service/history/command_checker.go
Comment (yycptt) [inline service/history/command_checker.go:1]: I think we also need to validate workflowID in commands?

File: common/searchattribute/encode_value.go [hunk 1]
```diff
@@ -72,16 +76,37 @@ func DecodeValue(
...
 
+func validateStrings(anyValue any, err error) (any, error) {
+	if err != nil {
+		return anyValue, err
+	}
+
+	// validate strings
+	switch value := anyValue.(type) {
+	case string:
+		if !utf8.ValidString(value) {
```
Comment (yycptt) [inline common/searchattribute/encode_value.go:97]: nit: can reuse common.ValidateUTF8String?

**Review by yycptt — APPROVED** [review]

File: common/searchattribute/encode_value.go (hunk 1 above)
Comment (yiminc) [reply common/searchattribute/encode_value.go:97]: That one is used to validate a field and require field name as input.

File: service/history/command_checker.go [hunk 2]
```diff
@@ -748,6 +752,10 @@ func (v *commandAttrValidator) validateStartChildExecutionAttributes(
 		return failedCause, serviceerror.NewInvalidArgument(fmt.Sprintf("WorkflowType on StartChildWorkflowExecutionCommand exceeds length limit. WorkflowId=%s WorkflowType=%s Length=%d Limit=%d Namespace=%s", wfID, wfType, len(wfType), v.maxIDLengthLimit, ns))
 	}
 
+	if !utf8.ValidString(wfType) {
+		return failedCause, serviceerror.NewInvalidArgument(fmt.Sprintf("WorkflowType %v is not a valid UTF-8 string", wfType))
+	}
```
[the remark this answers is not in the document]
Comment (yiminc) [reply service/history/command_checker.go]: sure

File: service/frontend/workflow_handler.go [hunk 3]
```diff
@@ -675,6 +673,9 @@ func (wh *WorkflowHandler) RespondWorkflowTaskCompleted(
 	if len(request.GetIdentity()) > wh.config.MaxIDLengthLimit() {
 		return nil, errIdentityTooLong
 	}
+	if err := common.ValidateUTF8String("Identity", request.GetIdentity()); err != nil {
+		return nil, err
+	}
```
[the remark this answers is not in the document]
Comment (yiminc) [reply service/frontend/workflow_handler.go]: no, will remove.

File: service/history/command_checker.go
Comment (yiminc) [reply service/history/command_checker.go:1]: Yes, added.

**Pull request merged**